.github/workflows/_deploy.yml

```diff
@@ -0,0 +1,49 @@
+---
+name: _deploy
+run-name: Deploy to ${{ inputs.environment }}
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+        description: "Environment to deploy to"
+      project-name:
+        type: string
+        required: true
+        description: "Name of the project to deploy"
+      is-dry-run:
+        type: boolean
+        default: true
+      version:
+        type: string
+        required: true
+
+jobs:
+  deploy:
+    name: Deploy ${{ inputs.project-name }} to ${{ inputs.environment }}
+    runs-on: ubuntu-22.04
+    environment: ${{ inputs.environment }}-${{ inputs.project-name }}
+    concurrency: ${{ inputs.environment }}-${{ inputs.project-name }}
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a  # v3.0.2
+        with:
+          name: ${{ inputs.project-name }}.zip
+
+      - name: Login to Azure
+        if: ${{ ! inputs.is-dry-run }}
+        uses: Azure/login@1f63701bf3e6892515f1b7ce2d2bf1708b46beaf  # v1.4.3
+        with:
+          creds: ${{ secrets.APP_SERVICE_SP }}  # From the environment.secrets
+
+      - name: Configure & Deploy App
+        if: ${{ ! inputs.is-dry-run }}
+        run: |
+          az webapp config appsettings set -g ${{ secrets.AZURE_RESOURCE_GROUP_NAME }} \
+            -n ${{ secrets.AZURE_APP_NAME }} --slot staging \
```

Comment:
@joseph-flinn API and Admin runs on different apps, so we need different variables based on project name

Reply:
These should be handled in the GitHub Environment. The secret name will be the same for each